### ISSUE TITLE
Budget: only the last budget directive is used

### ISSUE THREAD

**Issue by corani** [aumayr/fava#320]
It seems only the last budget statement for a category is considered. If I set a budget from 2015-01-01 and another for 2016-01-01, I only see the latter.

For example, consider:
```
2016-01-01 custom "budget" Expenses:Groceries "weekly" 100.00 XYZ
2016-04-01 custom "budget" Expenses:Groceries "weekly" 120.00 XYZ
2016-07-01 custom "budget" Expenses:Groceries "weekly" 110.00 XYZ
```

When I view the generated budgets based on these rules, they start from 2016-07-01 with a weekly budget of 110. The other rules are ignored.

**Comment by aumayr** [comment]
@corani I can't quickly reproduce this… but there was a bug involving multiple currencies, that was fixed in f746cff55d1362ebe21d8c2b4ff191fa74d14be2. 

Does this still occur? If yes: Can you provide a small test-beancount-file?

**Comment by corani** [comment]
It indeed appears to be fixed now. However, I did find another issue:

```
2016-01-01 custom "budget" Expenses:Groceries "weekly" 100.00 CNY
2016-06-01 custom "budget" Expenses:Groceries "weekly"  10.00 EUR
```

In this case, when the second budget statement becomes effective, the first one is cleared. I'd expect that from that day the effective budget would be `100.00 CNY + 10.00 EUR`, since I didn't reset the original.
